storage_json.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	model "github.com/golang-bristol/beer-model"
+	"github.com/nanobox-io/golang-scribble"
+)
+
+// StorageJSON is the data storage type using JSON file
+type StorageJSON struct {
+	db *scribble.Driver
+}
+
+var (
+	collectionBeer   = "beers"
+	collectionReview = "reviews"
+)
+
+func newStorageJSON(location string) (*StorageJSON, error) {
+	var err error
+
+	stg := new(StorageJSON)
+
+	stg.db, err = scribble.New(location, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return stg, nil
+}
+
+// SaveBeer insert or update beers
+func (s *StorageJSON) SaveBeer(beers ...model.Beer) error {
+	for _, beer := range beers {
+		var err error
+		var resource = fmt.Sprintf("%d", beer.ID)
+
+		beersFound, err := s.FindBeer(model.Beer{ID: beer.ID})
+		if err != nil {
+			return err
+		}
+
+		if len(beersFound) > 0 {
+			err = s.db.Delete(collectionBeer, resource)
```

Comment:
Then again,  this is more like a limitation on `golang-scribble` which has no update method, so deleting and writing was the quick option for this solution's proposal.

The idea for the `Save` function is to "upsert" data, but I think we could separate in two functions `Insert` and `Update`, so when trying to insert an already existing beer, it would return an error, and when trying to update would return an error when beer doesn't exist.

Reply:
That's fine if golang-scribble doesn't support update, guess we don't need to worry about it for now. I'd just make save return an error if the beer already exists rather than overwrite it - less surprises down the road that way :)

Reply:
sounds good.